routers/user/setting_openid.go

```diff
@@ -40,6 +40,12 @@ func SettingsOpenID(ctx *context.Context) {
 
 // SettingsOpenIDPost response for change user's openid
 func SettingsOpenIDPost(ctx *context.Context, form auth.AddOpenIDForm) {
+
+	if !setting.Service.EnableOpenIDSignUp {
```

Comment:
I think there should be check for `EnableOpenIDSignIn` not for `EnableOpenIDSignUp` that is for new user registration